pkg/region/cache.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains the read through cache implementation for
+// this package.
+package region
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// RegionCachePath is the $HOME/<RegionCachePath> location of the regionpicker cache
+	RegionCachePath = filepath.Join(".outreach", ".cache", "box")
+	// RegionCacheFile is the name of the regionpicker cache file
+	RegionCacheFile = "regions.json"
+)
+
+// cache is the package specific cache instance used. This is a global variable to
+// ensure that this is go-routine safe.
+var cache = &cacheStore{}
+
+// cacheEntry is information on a cloud's cache entry
+type cacheEntry struct {
+	// Duration is how long it took to talk to this region at LastUpdatedAt
+	Duration time.Duration `json:"duration"`
+
+	// LastUpdatedAt is the last time this file was updated. UTC.
+	LastUpdatedAt time.Time `json:"last_updated_at"`
```

Comment:
I suggest converting this to ExpiresAt instead.  It is a bit subtle but the main benefit is that you can implement negative caching more naturally eventually (so if a ping failed for a region, you can set duration to be time.10*Hour but set ExpiresAt to be 5 seconds to force it to be cached for far less time).

It doesn't apply in this specific case as much I think but it is worth sticking to that pattern for things that expire.